core/transaction/input.go

```diff
@@ -19,6 +19,9 @@ type TxnInput struct {
 	ReferTxOutputIndex uint16
 }
 
+// MaxRefIndex is the max of TxnInput.ReferTxOutputIndex
+const MaxRefIndex = ^uint16(0)
```

Comment:
Should we use math.MaxUint16 instead?

Reply:
I suggest don't instead of MaxUint16. Because we can't assumed type of ReferTxOutputIndex always is uint16.

Reply:
My expected solution is 
```
type IdxType uint16
ReferTxOutputIndex IdxType
MaxRefIndex = 1<<(sizeof(IdxType)*8) -1
```
But golang can't unsafe.Sizeof a Type without instantiated...
Have you any suggestions?

Reply:
I think it's enough to define MaxRefIndex as int, e.g.
const MaxRefIndex = 12345
because it's always compared with int (return of len)

Reply:
Change the type of ReferTxOutputIndex is backward incompatible update...

Reply:
@yilunzhang I submitted a new change 
182ce51, type define MaxRefIndex as int with initialized via math.MaxUint16.